### DISCUSSION TITLE
Where is the files subdirectory, when it isn't in the repo?

### DISCUSSION THREAD

**Discussion by Rod-at-DOH** [community/community#132434] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I've got a repo with SQL code in it. Months ago, I created an issue and pointed to a Word document. Time and other urgent priorities have pulled me away, so I've not been able to do anything with the issue until now. I saw the link to the Word document in the issue, and I clicked on it. It downloaded it to my new PC fine. However, the weird thing is the Word document is in a folder named *files*, but there's no *files* folder in the repo. Instead, the path looks something like this:

> files/14232192/MyWordDocument.docx

How can there be a folder named *files* in the repo, when there's no folder named *files* in the repo? What in heck is going on?

**Comment by Bram1903** [comment, marked as answer]
Hey there,

When you attach a file to a GitHub comment, GitHub stores the file and generates a link to it. This file is not actually stored in your repository, but the link format can make it appear as if it is part of the repository's directory structure. This can be confusing, as the file path seems to indicate the file is within the repository.

For example, a path like `files/14232192/MyWordDocument.docx` suggests that there is a `files` folder in your repo, but it's actually just a link GitHub created. This behavior has led to some confusion, especially when users attach files (like malware) to issues in well-known organizations, such as the National Security Agency, giving the false impression that these files are part of the organization's repository. This led people to believe a file was safe to download as they assumed it was part of a well-known organization. The biggest issue was that one didn't have to actually post the comment; simply uploading the file to the comment already created the link. Maintainers couldn't moderate this either, as they had no control or visibility over these files.

Unharmful example: https://github.com/NationalSecurityAgency/ghidra/assets/118324883/b8209e95-1bb7-4c1c-875b-8cceed44c3a1
The above link looks like a file that is part of the `ghidra` repository under the `NationalSecurityAgency` organization, even though it isn't.

GitHub has since addressed this issue by changing how it resolves file links in comments. Now, when you attach a file, it uses a different URL structure (such as `user-content`) to avoid this confusion and improve security.

See [GitHub Bug Used to Infect Game Hackers With Lua Malware](https://research.openanalysis.net/github/lua/2024/03/03/lua-malware.html), if you are further interested in how this worked!

I hope this answers your question.

Kind regards,  
Bram

**Reply by Rod-at-DOH** [reply]
Hi Bram, yes this does answer my question. I'm starting to use a *docs* folder at the root level of my repos, then putting in Word, Excel, whatever documents into that folder, to help me remember what the repo's about, the requirements users have asked for, etc. I presume that the *docs* folder will not be similarly treated like GitHub has done for other files attached to repos.

**Reply by Bram1903** [reply]
Yeah indeed, using a docs directory will work just fine, and won't cause any issue!
